includes/models/model.llms.lesson.php

```diff
@@ -720,190 +719,216 @@ public function has_content() {
 	 *
 	 * @since 1.0.0
 	 * @since 3.24.0
 	 * @since 4.4.0 Improve query so that unpublished siblings do not break expected results.
-	 * @since [version] Use a numeric comparison for the next position meta query.
+	 * @since [version] Use a numeric comparison for the previous position meta query.
+	 * @since [version] Refactor to use helper method `get_sibling()`.
 	 *
-	 * @return false|int ID of the next lesson, if any, `false` otherwise.
+	 * @return false|int WP_Post ID of the next lesson or `false` if one doesn't exist.
 	 */
 	public function get_next_lesson() {
 
-		// phpcs:disable WordPress.DB.SlowDBQuery.slow_db_query_meta_key
-		// phpcs:disable WordPress.DB.SlowDBQuery.slow_db_query_meta_query
-		// phpcs:disable WordPress.WP.PostsPerPage.posts_per_page_posts_per_page
+		return $this->get_sibling( 'next' );
+
+	}
+
+	/**
+	 * Get previous lesson ID
+	 *
+	 * @since 1.0.0
+	 * @since 3.24.0 Unknown.
+	 * @since 3.24.0 Unknown.
+	 * @since 4.4.0 Improve query so that unpublished siblings do not break expected results.
+	 *              Use strict comparisions where needed.
+	 *              Make sure to always return `false` if no previous lesson is found.
+	 * @since [version] Use a numeric comparison for the previous position meta query.
+	 * @since [version] Refactor to use helper method `get_sibling()`.
+	 *
+	 * @return false|int WP_Post ID of the previous lesson or `false` if one doesn't exist.
+	 */
+	public function get_previous_lesson() {
+
+		return $this->get_sibling( 'prev' );
+
+	}
+
+	/**
+	 * Retrieve the sibling lesson in a specified direction.
+	 *
+	 * @since [version]
+	 *
+	 * @param string $direction Direction of navigation. Accepts either "prev" or "next".
+	 * @return false|int WP_Post ID of the sibling lesson or `false` if one doesn't exist.
+	 */
+	protected function get_sibling( $direction ) {
+
+		$lesson = $this->get_sibling_lesson_query( $direction );
+
+		// No lesson found within the section, look within the sibling section.
+		if ( ! $lesson ) {
+			$lesson = $this->get_sibling_section_query( $direction );
+		}
+
+		return $lesson;
+
+	}
+
+	/**
+	 * Performs a query to retrieve a sibling lesson in the specified direction
+	 *
+	 * This method tries to locate a sibling lesson in the next or previous position.
+	 *
+	 * It *does not* account for lessons in a sibling section. For example, if the lesson
+	 * is the last lesson in a section this function will *not* locate the first lesson
+	 * in the course's next section. For this reason this function should not be relied upon
+	 * alone.
+	 *
+	 * @since [version]
+	 *
+	 * @param string $direction Direction of navigation. Accepts either "prev" or "next".
+	 * @return false|int WP_Post ID of the sibling lesson or `false` if one doesn't exist.
+	 */
+	private function get_sibling_lesson_query( $direction ) {
+
+		$curr_position = $this->get_order();
+
+		// First cannot have a previous.
+		if ( 0 === $curr_position && 'prev' === $direction ) {
```

Comment:
```suggestion
		if ( 1 === $curr_position && 'prev' === $direction ) {
```
the minimum `order` value is 1 not 0.